Breaking change in 3.1.8: using config.import with gives [//] is not a valid HTTP URL

**Describe the bug**

After upgrading from 3.1.7 to 3.1.8 using config.import with optional and using default "://" gives a "[//] is not a valid HTTP URL error".

Stacktrace:
```
Caused by: java.lang.IllegalArgumentException: [//] is not a valid HTTP URL
	at org.springframework.web.util.UriComponentsBuilder.fromHttpUrl(UriComponentsBuilder.java:316)
```

**Sample**
Using this config in application.yml:
```
spring:
    config.import:
      - optional:vault://
      - optional:configserver:${SPRING_CLOUD_CONFIG_URI://}
```

As reference this should still be valid use: 
[Conditionally enable/disable Vault Configuration](https://docs.spring.io/spring-cloud-vault/docs/current/reference/html/config-data.html#vault.configdata.location.optional)
In some cases, it can be required to launch an application without Vault. You can express whether a Vault config location should be optional or mandatory (default) through the location string:

optional:vault:// (default location)

See: https://docs.spring.io/spring-cloud-vault/docs/current/reference/html/config-data.html

Using these dependencies:
```
<spring-boot.version>2.7.13</spring-boot.version>
<spring-cloud.version>2021.0.8</spring-cloud.version>

```

--- a/spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServerConfigDataResource.java
+++ b/spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServerConfigDataResource.java
@@ -123,17 +123,27 @@ public class ConfigServerConfigDataResource extends ConfigDataResource {
 	}
 
 	private boolean uriEqual(String thisUriString, String thatUriString) {
-		UriComponents thisUri = UriComponentsBuilder.fromHttpUrl(thisUriString).build();
-		UriComponents thatUri = UriComponentsBuilder.fromHttpUrl(thatUriString).build();
-		return Objects.equals(thisUri.getHost(), thatUri.getHost())
-				&& Objects.equals(thisUri.getPort(), thatUri.getPort())
-				&& Objects.equals(thisUri.getPath(), thatUri.getPath());
+		try {
+			UriComponents thisUri = UriComponentsBuilder.fromHttpUrl(thisUriString).build();
+			UriComponents thatUri = UriComponentsBuilder.fromHttpUrl(thatUriString).build();
+			return Objects.equals(thisUri.getHost(), thatUri.getHost())
+					&& Objects.equals(thisUri.getPort(), thatUri.getPort())
+					&& Objects.equals(thisUri.getPath(), thatUri.getPath());
+		}
+		catch (Exception e) {
+			return Objects.equals(thisUriString, thatUriString);
+		}
 	}
 
 	private int urisHashCode(String[] uris) {
 		return Arrays.stream(uris).mapToInt(uriString -> {
-			UriComponents uriComponents = UriComponentsBuilder.fromHttpUrl(uriString).build();
-			return Objects.hash(uriComponents.getHost(), uriComponents.getPath(), uriComponents.getPort());
+			try {
+				UriComponents uriComponents = UriComponentsBuilder.fromHttpUrl(uriString).build();
+				return Objects.hash(uriComponents.getHost(), uriComponents.getPath(), uriComponents.getPort());
+			}
+			catch (Exception e) {
+				return Arrays.hashCode(uris);
+			}
 		}).sum();
 	}
 

--- a/spring-cloud-config-client/src/test/java/org/springframework/cloud/config/client/ConfigServerConfigDataResourceTests.java
+++ b/spring-cloud-config-client/src/test/java/org/springframework/cloud/config/client/ConfigServerConfigDataResourceTests.java
@@ -318,4 +318,15 @@ class ConfigServerConfigDataResourceTests {
 		assertThat(r1.hashCode()).isNotEqualTo(r2.hashCode());
 	}
 
+	@Test
+	void testInvalidUri() {
+		ConfigClientProperties r1Properties = new ConfigClientProperties();
+		r1Properties.setUri(new String[] { "//" });
+		ConfigServerConfigDataResource r1 = new ConfigServerConfigDataResource(r1Properties, true, null);
+		ConfigClientProperties r2Properties = new ConfigClientProperties();
+		r2Properties.setUri(new String[] { "//" });
+		ConfigServerConfigDataResource r2 = new ConfigServerConfigDataResource(r2Properties, true, null);
+		assertThat(r1).isEqualTo(r2);
+	}
+
 }

